packages/dds/tree/src/index.ts

```diff
@@ -32,12 +32,12 @@ export {
 } from "./core/index.js";
 export { type Brand } from "./util/index.js";
 
-export {
-	type Listeners,
-	type IsListener,
-	type Listenable,
-	type Off,
-} from "./events/index.js";
+export type {
```

Comment:
Yes, it is just to avoid breaking changes. I'll open a follow-up work item to remove them

Reply:
Removing these will be a breaking change on public API surface so it'll have to wait until 3.0, I think (?) but we'll have to deprecate them somehow before that, no? Should we do something like importing them as `Listeners_base` or similar in this package, and exporting them from here as a new type with its own docs and deprecation notice? I.e. something like (double check it, not sure it works as written):

```
import { Listeners as Listeners_base } from "@fluidframework/core-interfaces";

/**
 * @inheritDoc <something here, I don't recall the syntax>
 * @ deprecated This type has moved to `@fluidframework/core-interfaces` and will be removed from here in a future release.
 * If you need to reference it directly, do it from `@fluidframework/core-interfaces`.
 */
export type Listeners = Listeners_base;
```

Reply:
I believe this is not considered breaking since it's re-exported from here as well, and there are no typetest breaks in tree.

Reply:
Right, it's not breaking now, but when we remove them from here it will be breaking, and before we can do that they should be marked as deprecated here. So I'm suggesting we deprecate them here now (while keeping them not-deprecated in core-interfaces).

Reply:
This sounds reasonable, but then we probably also need to export the (non-deprecated) API from the `fluid-framework` package, as we don't want 3p customers directly importing from client-utils and otherwise they won't have a non-deprecated option available.

Might be better to file this deprecation as a task for now as touching `fluid-framework` is probably the final step, but agree that we need to deprecate well in advance of 3.0.

Reply:
ADO: https://dev.azure.com/fluidframework/internal/_workitems/edit/25440

Another concern could be aliasing in core-interfaces: https://github.com/sonalideshpandemsft/FluidFramework/pull/41. 

Would that be a problem? Is it okay for aliasing to happen in tree but that will cause typetest breaks?

Reply:
Wrong link about the aliasing bit? Not sure what the concern is there.

Reply:
Aliasing in core-interfaces: https://github.com/microsoft/FluidFramework/pull/23180/commits/04274d87f4d8a6a67aba70fd1a353341739bbab3

Reply:
I'm actually not sure why Intellisense picks up on the deprecation notice and nothing else. For what it's worth, our API docs _will not_ understand this. Unfortunately, I think you would need to do what @alexvy86 suggested to make everything work out correctly. E.g.

```ts
import { Foo as FooBase } from "a";

/**
 * Foo description.
 * @deprecated ...
 */
export type Foo = FooBase;
```

I don't believe you can use `{@inheritDoc}` in this case, unfortunately, since we're introducing local documentation (the deprecated tag).

Are all of these re-exports type-only? Or are any of them classes or anything? Making the re-export of a class would require subclassing to remain compatible.

Reply:
They are all type-only.